axelor-human-resource/src/main/java/com/axelor/apps/hr/service/EmployeeDashboardServiceImpl.java

```diff
@@ -0,0 +1,283 @@
+
+      if (BigDecimal.ZERO.compareTo(expectedWorkingDuration) == 0) {
+        continue;
+      }
+      if (spentTimeInDay.compareTo(expectedWorkingDuration) >= 0) {
+        atWork++;
+      } else {
+        notAtWork++;
+      }
+    }
+
+    Map<String, Object> dataMap = new HashMap<>();
+    dataMap.put(AT_WORK, atWork);
+    dataMap.put(NOT_AT_WORK, notAtWork);
+
+    return List.of(dataMap);
+  }
+
+  @Override
+  public List<Map<String, Object>> getExtraHrsData() throws AxelorException {
+    Employee employee = getEmployee();
+    Period period = getCurrentPeriod();
+
+    List<ExtraHoursLine> extraHoursList =
+        extraHoursRepo
+            .all()
+            .filter(
+                "self.extraHours.employee = :employee AND self.date >= :fromDate AND self.date <= :endDate")
+            .bind("employee", employee)
+            .bind("fromDate", period.getFromDate())
+            .bind("endDate", period.getToDate())
+            .fetch();
+    List<Map<String, Object>> extraHrsData = new ArrayList<>();
+    for (ExtraHoursLine extraHrsLine : extraHoursList) {
+      Map<String, Object> map = new HashMap<>();
+      map.put("extraHrId", extraHrsLine.getExtraHours().getId());
+      map.put("date", extraHrsLine.getDate());
+      map.put("duration", extraHrsLine.getQty());
+      map.put("description", extraHrsLine.getDescription());
+      extraHrsData.add(map);
+    }
+    return extraHrsData;
+  }
+
+  protected Map<LocalDate, BigDecimal> getTimeSpentPerDay(Employee employee, Period period) {
+    List<TimesheetLine> timesheeLineList =
+        timesheetLineRepo
+            .all()
+            .filter(
+                "self.timesheet.employee = :employee AND self.timesheet.statusSelect = :statusSelect AND self.date >= :fromDate AND self.date <= :endDate")
+            .bind("employee", employee)
+            .bind("statusSelect", TimesheetRepository.STATUS_VALIDATED)
+            .bind("fromDate", period.getFromDate())
+            .bind("endDate", period.getToDate())
+            .fetch();
+
+    return timesheeLineList.stream()
+        .collect(
+            Collectors.groupingBy(
+                TimesheetLine::getDate,
+                Collectors.reducing(
+                    BigDecimal.ZERO, TimesheetLine::getHoursDuration, BigDecimal::add)));
+  }
+
+  @Override
+  public Period getCurrentPeriod() throws AxelorException {
+    return Beans.get(PeriodService.class)
+        .getActivePeriod(
+            LocalDate.now(), AuthUtils.getUser().getActiveCompany(), YearRepository.TYPE_PAYROLL);
+  }
+
+  @Override
+  public Employee getEmployee() throws AxelorException {
+    Employee employee =
+        Optional.ofNullable(AuthUtils.getUser()).map(User::getEmployee).orElse(null);
+    if (employee == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get("No employee found for connected user"));
```

Comment:
Please put this exception message inside HumanResourceExceptionMessage.

Also please move the method `getEmployee` to a more appropriate service (I think EmployeeService is fine) and rename the method to `getConnectedEmployee`